### ISSUE TITLE
[Github] Close old issues that are no longer applicable

### ISSUE THREAD

**Issue by lexidor** [facebook/hhvm#8573]
The following issues are stale and keeping the issue count on hhvm high:

| Issue reference | Verdict |Reason for closing |
| -----------------: | :------: | --------------------------|
| #1066 | Stale |This refers to the performance of HHVM back in 2013. |
| #1447 | PHPism? | #7926 (unmerged) addressed this issue. Unsure how to feel about modifying a file during a request. This does sound like a PHPism to me. |
| #1605 | Implementation detail of PHP | You should not be relying on the order of constants in which constants are defined in get_defined_constants. |
| #1589 | Famously unmaintained | It looks like the last non-cleanup commit in the code was about 22 months ago. (Jan 2014) |
| #1669 | PHPism | All hack type hints are strictly adhered to. |
| #2060 | Programmer Error | Registering a private function to a callback based api is not the fault of the api, but rather a programmer error. |
| #2108 | Deprecated | From a conversation with a Facebook employee -> "filters are gone, wrappers that can re-enter the VM are gone." Feels like it applies here too. |
| #2227 | Fixed | I can see an [HHI](https://github.com/facebook/hhvm/blob/master/hphp/hack/hhi/weakref.hhi) for WeakRef, so it exists now. |
| #2260 | Fixed | Tried in nightly and it works there. Assume that this works everywhere else too. |
| #2268 | PHP feature unsupported | You can't use `ArrayAccess<_, _>` anyhow since hack will tell you that you are using a subscript operator on something that is not a `KeyedContainer<_, _>` `$foo = new Foo(); $foo['a'] = 'b';``Typing[4005] This is not an object of type KeyedContainer, this is an object of exactly the class Foo`. |
| #2406 | hhvm-autoload, HH\fact_parse, and hhast exist | It seems like `@fredemmott` was trying to do what hhvm-autoload is doing now. I presume that hhvm is providing the required functions and hooks to perform this, else how would hhvm-autoload do it? |
| #2560 | Works? | I tried the example from [PHP docs on PHP authentication](https://www.php.net/manual/en/features.http-auth.php) and it worked. It is unclear to me which fields may still be missing. I was able to see most of them using `hhvm -m server` and vardumping the keys of `$_SERVER`. |
| #2692 | Fixed? | My own website is serving some static `.json` files just fine. That might be because of how my op configured it though. It doesn't seem like this is an issue anymore. |
| #2730 | Can be invoked using parens | [blog](https://hhvm.com/blog/2019/06/18/hhvm-4.10.0.html) mentions `($this->callable_property)(...$args)` |
| #2926 | Untyped hack | `require_once` (like `echo` and friends) is an untyped statement. This is expected behavior. You should not be requiring things yourself if you can help it. Autoloading is baked into the core of hhvm. The only thing that really needs to be required these days is `/vendor/bin/autoload.hack`. |
| #2957 | Sample code unsupported | It is hard to figure out what sql lib was used. It might have been mysqli. Even if it was something else, the given sample code is php that exploits many deprecated features. I don't think that this is something that we can realistically still use. |
| #2976 | Stub | https://github.com/facebook/hhvm/issues/2976#issuecomment-46467245 |
| #3015 | HHI change | `call_user_func_array` takes `Container<_>`, stdClass is stopped at typecheck time.
| #3193 | PHPism | We are strict about the difference between static methods and instance methods. |
| #3273 | Updated | This describes the current behavior of `// strict`. |

Okay, this took waaay to long already. Will flag a couple of issues every time I create a new one. I want the issue count to slowly go down to something that is more manageable.

**Comment by ecreeth** [comment]
| Issue    |  Reason                                                                                            |
|-----------:|----------------------------------------------------------------------------------------:|
| #6954   | PHP related                                                                                      |

**Comment by lexidor** [comment]
@ecreeth Could you explain your rationale for #3874? Is this okay to close because this behavior is also correct, but just different from PHP? Or are we merely closing this because of the PHP comparison?

**Comment by ecreeth** [comment]
@lexidor This is due to the comparison with php and the year it was created

**Comment by lexidor** [comment]
The quirks that were mentioned in the issue are still accurate to this day. The example does not work out of the box, since `ReflectionFunction::getDefiningFunction()` is not a thing, you need `ReflectionFunction::getDeclaringFunction()`.

## HHVM + Hack 4.38.0-dev (rel)
```
bool(false)
NULL
object(ReflectionClass) (2) {
  ["name"]=>
  string(27) "Closure$Ship::getClosure;10"
  ["obj":"ReflectionClass":private]=>
  NULL
}
```

## PHP + php 7.4.1 (cli)
```
bool(true)
object(ReflectionClass)#2 (1) {
  ["name"]=>
  string(4) "Ship"
}
object(ReflectionClass)#2 (1) {
  ["name"]=>
  string(4) "Ship"
}
```

I don't speak Reflectionista and I don't know if anyone uses this / cares.
It is worth noting that `ReflectionFunction::getClosureScopeClass()` does not exist in the hhi's.

```HACK
<?hh // partial
//<?php

function main(): void {
  $ship = new Ship();
  $rf = new ReflectionFunction($ship->getClosure());
  $rf = $rf->getParameters()[0]->getDeclaringFunction();
  \var_dump($rf->isClosure());
  \var_dump($rf->getClosureScopeClass());
  \var_dump($rf->getDeclaringClass());
}

class Ship {
  public function getClosure() {
    return function(string $canon_to_fire) {
      return false;
    };
  }
}

main();
```

**Comment by lexidor** [comment]
I am going over old issues on this repository, to see which ones apply to the current versions of hhvm.

We've come full circle.
I am closing my own issue about needing to close issues.

To all those issues that are on the list above, that I have not closed:
Sorry for marking your issue as one that needs to be closed.
With my far more conservative view on the issues list, I have closed some, but not all of them.
This issue was not made with the level of care that is required for working on open source.

I mainly focused on closing issues that relied on features that were removed from hhvm or relied on php mode.

I am updating the issue title of this issue to three underscores in order to not swing the discussion on whether to keep your issue open.
